libraries/AP_NavEKF3/AP_NavEKF3_MagFusion.cpp

```diff
@@ -264,7 +264,9 @@ void NavEKF3_core::SelectMagFusion()
     if ((effectiveMagCal == MagCal::GSF_YAW || !use_compass()) &&
         effectiveMagCal != MagCal::EXTERNAL_YAW &&
         effectiveMagCal != MagCal::EXTERNAL_YAW_FALLBACK) {
-        if (!yawAlignComplete) {
+
+        // because this type of reset event is not as time critical, require a continuous history of valid estimates
+        if (!yawAlignComplete && EKFGSF_yaw_valid_count >= GSF_YAW_VALID_HISTORY_THRESHOLD) {
```

Comment:
EKFGSF_yaw_valid_count is only incremented when gpsDataToFuse is true. Do we want to support the GSF for vision based position systems?

Reply:
If they have velocity data, then we could, but  given the estimated yaw yaw would be relative to vision earth frame and the vision systems already provide a yaw of body wrt that frame fo reference, there doesn't seem to be much use for it at the moment. I would prefer to tackle that as separate PR with test support from the vision user community.